odoo/models.py

```diff
@@ -2596,6 +2596,16 @@ def _parent_columns_exist(self):
     def _create_parent_columns(self):
         self._cr.execute('ALTER TABLE "%s" ADD COLUMN "parent_left" INTEGER' % (self._table,))
         self._cr.execute('ALTER TABLE "%s" ADD COLUMN "parent_right" INTEGER' % (self._table,))
+        # Add unique constraint on the values of parent_left and parent_right to
+        # prevent race conditions during MPTT updates in record creation.
+        # /!\ Do not change the index names, as they are matched to retry
+        # transactions that failed because of them.
+        query = """ CREATE UNIQUE INDEX "{t}_parent_store_index0" ON "{t}"
```

Comment:
1. This will make the point 2. more probable...
2. You have to explain me how to truncate the model name without taking the risk of a collision...